Adds Shipping Methods to Checkout Page

supernova-at opened the pull request:
<!--
Before submitting this pull request, please make sure you have read our Contribution Guidelines and your PR meets our contribution standards:
https://github.com/magento/pwa-studio/blob/master/.github/CONTRIBUTING.md

Please fill out as much information as you can about your PR to help speed up the review process.
If your PR addresses an existing GitHub Issue, please refer to it in the title or Additional Information section to make the connection.

We may ask you for changes in your PR in order to meet the standards set in our Contribution Guidelines. PRs that do not comply with our guidelines may be closed at the maintainers' discretion.

Feel free to remove this section before creating this PR. Thank you for your contribution!
-->

## Description

This PR adds shipping methods to the Checkout Page.

**For reviewer consideration:**

1. I purposefully kept all the logic work in the `useShippingMethod` talon, as opposed to creating talons for each subcomponent. I made this choice because it keeps all of the logic in one place, it's easier to separate the component view from the talon for other implementations of the view, and it's easier to maintain. The downside is that I had to do some additional prop threading.
1. Speaking of prop threading, I purposefully did not create a `ShippingMethod` context because there wasn't enough of a maintenance concern to justify it. Also for consistency with other components, few of which create their own contexts.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here by replacing ISSUE_NUMBER with your actual issue number. -->
<!--- Using the above wording causes Github to automatically close the issue on merge. -->
Closes PWA-179.

## Acceptance 
<!-- The people and processes this pull request needs before it is merged. -->
<!-- These fields are not required when opening the pull request, but they -->
<!-- should be populated after code review. -->
### Verification Stakeholders
<!-- People who must verify that this solves the attached issue. -->

@soumya-ashok 

### Specification
<!-- Changes to `upward-spec` and/or `upward-js` packages must be reviewed -->
<!-- by `UPWARD-PHP` maintainers to ensure continued compatibility -->

### Verification Steps
<!-- Please describe in detail how a reviewer can verify your changes, -->
<!-- OR how you will demonstrate the changes to the stakeholder(s). -->

#### No Shipping Address

📝 This shouldn't really ever happen in production but since the Shipping Address component hasn't been completed yet, it is possible to get to Shipping Methods without first having input a Shipping Address.

1. Clear your site data (`Developer Tools > Application > Clear storage > Clear site data`)
1. Hard reload the home page
1. Add an item to your cart
1. Navigate to the `/checkout` page
1. Click "Continue to shipping method" button in the "Shipping Address" section
1. Verify that the Shipping Method UI is:

![Screen Shot 2020-03-25 at 1 27 10 PM](https://user-images.githubusercontent.com/13182778/77572079-5fd57680-6e9c-11ea-97c0-b7d097e61f2a.png)

### Shipping Address Has No Shipping Method

📝 This often occurs during guest checkout.

1. Continue directly from or repeat the steps above
1. Open the mini cart
1. Click "Checkout" and add a shipping address
1. Refresh the `/checkout` page<sup>1</sup>
1. See that there is a brief loading spinner while shipping methods are fetched
1. Verify that the Shipping Method UI shows and pre-selects the lowest cost available shipping method:

![Screen Shot 2020-03-25 at 1 45 48 PM](https://user-images.githubusercontent.com/13182778/77573626-f4d96f00-6e9e-11ea-85dc-97314f4b3ca2.png)

### Shipping Address Has A Shipping Method

1. Clear your site data (`Developer Tools > Application > Clear storage > Clear site data`)
1. Hard reload the home page
1. Add an item to your cart
1. Navigate to the `/checkout` page
1. Open the mini cart
1. Click "Checkout" 
1. Add a shipping address
1. Add a shipping method
1. Refresh the `/checkout` page<sup>1</sup>
1. Verify that the Shipping Method UI shows and pre-selects the previously selected shipping method:

![Screen Shot 2020-03-25 at 1 45 48 PM](https://user-images.githubusercontent.com/13182778/77573626-f4d96f00-6e9e-11ea-85dc-97314f4b3ca2.png)

<sup>1</sup> It would be nice if the page didn't have to reload to pick up the changes, but we don't have any patterns for that yet.

### "Done" View

1. Click the "Continue to Payment Information" button
1. Verify that the Shipping Method UI goes into done / read-only mode:

![Screen Shot 2020-03-25 at 1 48 49 PM](https://user-images.githubusercontent.com/13182778/77573914-631e3180-6e9f-11ea-8b0f-73389afb3881.png)

### Update

1. From the "Done" view, click anywhere inside the card
1. Verify that an update modal appears (on desktop), or the right drawer appears (mobile)
1. Verify you can close / cancel the update by clicking the "X" or the "Cancel" button
1. Verify you can update your selection by clicking the "Save" button

## Screenshots / Screen Captures (if appropriate)

## Checklist
<!--- Go over all the following points, and make sure you've done anything necessary -->
* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

soumya-ashok commented:
@supernova-at There should only be 2 heading styles. One when it is being edited and one when it is collapsed into a card. Not sure where the third is coming from.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editing.js:
```diff
@@ -0,0 +1,42 @@
+                    type="submit"
+                    disabled={isLoading || !hasShippingMethods}
+                >
+                    {'Continue to Payment Information'}
```
Going back to the edit view after submitting data presents a weird UX. The payment info and adjustment components are now displayed on the page but you have a "continue to payment information" button.

[![Image from Gyazo](https://i.gyazo.com/d150063b4b5eace38534e526b63cc921.gif)](https://gyazo.com/d150063b4b5eace38534e526b63cc921)

We may be able to work around this by changing this text if data has already been submitted (meaning we're past the shipping method step). Or we could always open "update" view in a modal.

@jimbo and @soumya-ashok and I had a discussion about this but I'm not sure if we documented it anywhere. I think there should be three states for each component.

1) An initial, unfilled form (currently the edit state)
2) a filled, "read only" view with an edit button
3) A modal with selections (update?)

Edit: After looking at the ticket, [mock #3](https://jira.corp.magento.com/secure/attachment/330380/Venia%20checkout%20-%20Mobile%20-%20Guest%20-%20Edit%20shipping%20method.png) is that "update" view.

supernova-at replied:
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/688480832b90327a5fa9116193c6fd9bf76c0297.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editing.js:
```diff
@@ -0,0 +1,42 @@
+                <Button
+                    priority="normal"
+                    type="submit"
+                    disabled={isLoading || !hasShippingMethods}
```
This should also be disabled if the checkout page `isUpdating` and while the submit mutation is in flight, otherwise I can spam the button and it'll queue a bunch of mutations!

[![Image from Gyazo](https://i.gyazo.com/b411aa15e1f6d53710bfcb3f6ff7b1fb.gif)](https://gyazo.com/b411aa15e1f6d53710bfcb3f6ff7b1fb)

supernova-at replied:
Fixed in https://github.com/magento/pwa-studio/pull/2280/commits/963ed01710348091380217d1054bd64d69613e82.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.js:
```diff
@@ -0,0 +1,66 @@
+};
+
+// ShippingRadios.propTypes = {
+
```
Ahem...

supernova-at replied:
This is an interesting point of discussion I suppose - I only included `propTypes` on the public API components not the child subcomponents (this comment has been removed).

Think that's valid?

sirugh replied:
I've been pretty bad about adding propTypes lately but I think adding them to the public api should be required. For child components it certainly doesn't hurt either :)

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.js:
```diff
@@ -0,0 +1,66 @@
+
+    const shippingRadios = shippingMethods.map(method => {
+        const label = (
+            <ShippingRadio
```
It's awesome that we can reuse some of the cart page components. That said, the `CartPage` shipping radios puts the selection price under the text whereas the mocks for the checkout page radios show the price to the right of the text. Personally I prefer we keep them identical on both cart and checkout but it is something we should raise with UX.

cc @soumya-ashok 

[![Image from Gyazo](https://i.gyazo.com/12c01cc4468a5dac4d22004f836dfbe0.png)](https://gyazo.com/12c01cc4468a5dac4d22004f836dfbe0)

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/done.css:
```diff
@@ -0,0 +1,48 @@
+}
+
+.heading {
+    font-weight: bold;
```
Jimmy would ask that we use number ie `600` instead of text alias for weight :)

supernova-at replied:
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/ee446ee8c75390b705b1c031e2ed875a1685da6e.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.js:
```diff
@@ -0,0 +1,66 @@
+
+    if (isLoading) {
+        return (
+            <LoadingIndicator>{`Loading Shipping Methods...`}</LoadingIndicator>
```
God we have so many loading indicators...

sirugh commented on packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js:
```diff
@@ -0,0 +1,158 @@
+    const [
+        fetchShippingMethods,
+        { data, error, loading: isLoadingShippingMethods }
+    ] = useLazyQuery(getShippingMethods);
```
Also don't forget your fetch policy for these.

supernova-at replied:
Thanks - I was thinking I wanted the default `cache-first`,  but folks could update shipping addresses and selected methods in other tabs, etc. so I went with `cache-and-network`: https://github.com/magento/pwa-studio/pull/2280/commits/2ad41c7d66af0e1b2a6815d69846c9512452aee7

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/done.js:
```diff
@@ -0,0 +1,76 @@
+import LoadingIndicator from '../../LoadingIndicator';
+import defaultClasses from './done.css';
+
+const editIconAttrs = {
```
Hmm I wonder if we should generalize the `done` state cards for use by other steps or auth view. Could take a title, content and `onEdit` callback. Would prevent everyone from rewriting their own versions of a done state.

supernova-at replied:
Yep. We decided that's out of scope here but I'll do a fast follow.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.js:
```diff
@@ -1,46 +1,90 @@
+                .set(
+                    displayStates.DONE,
+                    <Done
+                        isLoading={isLoadingSelectedShippingMethod}
```
Would this ever get used? The only time you'll render this state is when you KNOW that you have a selected shipping method.

supernova-at replied:
That ... is an excellent point 😄 

Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/377917ddd5d9fb8a0f7cfdec49bd36356389912e.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editing.css:
```diff
@@ -0,0 +1,15 @@
+.root {
+    display: grid;
+    grid-template-rows: min-content 1fr;
+    row-gap: 1em;
```
I think we should be using (r)elative em

supernova-at replied:
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/0115799684e2a042d57572f44b59328237c0298f.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/done.js:
```diff
@@ -0,0 +1,76 @@
+        <div className={classes.root}>
+            <button className={classes.button} onClick={showEditMode}>
+                <div className={classes.container}>
+                    <span className={classes.title_container}>
```
Per the discussion _somewhere_ I think we're leaning toward BEM with camel case ie `titleContainer`. I haven't checked the rest of your css but I would use it going forward.

supernova-at replied:
Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/8668ca69cc2a4fd82eb6b86ea658b24e0796366d.

The checkout page itself has `snake_case` names but I didn't change that here. All of my styles updated in the commit above.

tjwiebell commented:
@supernova-at: @sirugh suggested we align on some of our editing and detail styles. Since you're first in the hopper I'll let you work out these styles with UX, but wanted to throw out how my eyeballing from mocks turned out. I've got a PR up [here](https://github.com/magento/pwa-studio/pull/2285), but need to figure out how to kick off CI on a draft PR.

<img width="617" alt="Screen Shot 2020-03-26 at 8 17 28 AM" src="https://user-images.githubusercontent.com/462953/77651486-df198780-6f3a-11ea-988f-a16ab7acf92e.png">

![](https://gyazo.com/7a329ee64f5a1c092c25bc9e4defb399.gif)

supernova-at commented:
@tjwiebell Nice, yeah - it definitely occurred to me that a `Card` component or similar would be beneficial for re-use on this page. 

It feels a bit out of scope here honestly but I can do a fast follow and then we could all use that?

I'm thinking something like 

```jsx
<Card title={ your section's name }>
  <Editing>{ your contents here }</Editing>
  <Done>{ your contents here }</Done>
</Card>
```

or 

```jsx
<Card
  editing={your editing component here}
  done={your done component here}
  title={your section's name}
/>
```

The `Card` component would take care of styling the box, headers, adding the "edit" button in done mode, etc.

tjwiebell commented:
> @tjwiebell Nice, yeah - it definitely occurred to me that a `Card` component or similar would be beneficial for re-use on this page.
> ...

Agree with a fast follow, or something that gets added to the styleguide at some point and fixed in post like the rest of the styles. I just saw @jimbo post in Slack what these styles should be, so I think we're aligned. I'll just keep track of your PR, and assume whatever y'all arrive at is what I should apply too.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.js:
```diff
@@ -0,0 +1,66 @@
+
+const ShippingRadios = props => {
+    const {
+        hasShippingMethods,
```
Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/6d1d1a20510b28f328b5a834215c2f573c0be187.

Specifically to answer the question, `shippingMethods` defaults to an empty array, so components just check for `!shippingMethods.length` 👍

supernova-at commented on packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js:
```diff
@@ -0,0 +1,158 @@
+    const {
+        data: chosenShippingMethodData,
+        loading: isLoadingSelectedShippingMethod
+    } = useQuery(getSelectedShippingMethod);
```
👻 BOO!

😄 Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/f142a6b178ece53e7a577f1096002d7c9e8aabb8

And nice catch actually, I'm not sure how the `useQuery` was working before because I never passed it a `cartId` `variable`...

sirugh replied:
Spooky!

supernova-at commented on packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js:
```diff
@@ -0,0 +1,158 @@
+
+            setDisplayState(displayStates.DONE);
+
+            if (onSave) onSave();
```
Oh nice, thanks - I totally missed that the Checkout page had this functionality in place already 👍 

Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/963ed01710348091380217d1054bd64d69613e82.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editForm.js:
```diff
@@ -0,0 +1,69 @@
+            return (
+                <div className={classes.updateButtonContainer}>
+                    <Button onClick={handleCancelUpdate}>{'Cancel'}</Button>
+                    <Button priority="high" type="submit">
```
You should disable this button while the update is in flight.

supernova-at replied:
Good call. Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/3e47f4f62fdb55800ecd72a106b749f94b20f3e0.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.js:
```diff
@@ -0,0 +1,51 @@
+                    <EditForm
+                        handleCancelUpdate={handleCancelUpdate}
+                        handleSubmit={handleSubmit}
+                        mode={editFormModes.UPDATE}
```
@tjwiebell had a different approach to determining which state to display. He checks for the [existence of email](https://github.com/magento/pwa-studio/pull/2285/files#diff-628505350ad68f9d53340e7f89b5df0bR28) which indicates that a user has already submitted data. I'm not sure which I like better, but I just wanted to point out there is a minor difference here in the approach.

supernova-at replied:
I do a similar check for the [existence of a selected shipping method](https://github.com/magento/pwa-studio/blob/supernova/179_checkout_shipping_methods/packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js#L163-L177), but I just pass that answer to the modal in the form of a `mode` prop.

I'm not married to it but I thought it was a little more explicit in what the we wanted to show.

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.css:
```diff
@@ -0,0 +1,65 @@
+    opacity: 0;
+    overflow: hidden;
+    position: fixed;
+    transform: scale(1.15);
```
Why do we scale it up when not displayed? This causes a "shrinking" appearance instead of it just sliding in. I'll try to capture it in a gif. If you change this to `scale(1)` and the open translate to `translate(-50%, 0)` it just slides in without a scaling transform. I prefer this but I also don't know what you've discussed with UX so I'd just run it by them before having it merged.

[![Image from Gyazo](https://i.gyazo.com/fda4ec35089df9ac9c3998ef72cb1b2e.gif)](https://gyazo.com/fda4ec35089df9ac9c3998ef72cb1b2e)

supernova-at replied:
I just stole it from @tjwiebell , lol. I was curious about this too actually.

supernova-at replied:
While I liked the look of it scaling _up_, I removed scaling altogether (for consistency) in https://github.com/magento/pwa-studio/pull/2280/commits/12fea240f5712df18f060be17ecdd4c22c26f783.

sirugh replied:
![](https://i.imgflip.com/36naqm.png)

jimbo commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.js:
```diff
@@ -1,46 +1,114 @@
+                                handleSubmit={handleSubmit}
+                                isLoading={
+                                    isLoadingSelectedShippingMethod ||
+                                    isLoadingShippingMethods
```
Try creating another variable for this one.

supernova-at replied:
Resolved by a previous commit.

jimbo commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editForm.css:
```diff
@@ -0,0 +1,17 @@
+    display: grid;
+    grid-template-columns: min-content min-content;
+    column-gap: 1rem;
+    margin: 1.5rem auto 2rem auto;
```
```suggestion
    margin: 1.5rem auto 2rem;
```

supernova-at replied:
Removed by a previous commit.

jimbo commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editForm.css:
```diff
@@ -0,0 +1,17 @@
+
+.updateButtonContainer {
+    display: grid;
+    grid-template-columns: min-content min-content;
```
In general, `auto` is fine. There are uses for `min-content`, but they're rarer than you'd think. Consider reviewing the places you've used `min-content` and see if `auto` would be fine.

supernova-at replied:
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/dd1654f50b2ffd5c467bf4b03f8a2f6631f5dfd5.

jimbo commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/done.js:
```diff
@@ -0,0 +1,90 @@
+import Icon from '../../Icon';
+import defaultClasses from './done.css';
+
+const Done = props => {
```
This is a bit vague for a component name. Maybe iterate on it a bit?

supernova-at replied:
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/db303d8d61e68255ebbb4a8d07a9f3f5eab599fe.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.js:
```diff
@@ -0,0 +1,53 @@
+
+const UpdateModal = props => {
+    const {
+        handleCancelUpdate,
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/7383ca73f283c19a84b6eee294203c69eb9d1f7d.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.js:
```diff
@@ -0,0 +1,53 @@
+    );
+};
+
+export default UpdateModal;
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/7737a3e7c3a4f6b111331619501e940f73f3fd64.

devops-pwa-codebuild commented:
**Performance Test Results**

The following fails have been reported by WebpageTest

https://pr-2280.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.67

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.css:
```diff
@@ -0,0 +1,64 @@
+    visibility: hidden;
+    width: 100%;
+    z-index: 3;
+}
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/577f1a335eac9fd566f387945ff957ae07298680.

It's fun to understand grid! 😄

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.css:
```diff
@@ -0,0 +1,64 @@
+.header {
+    border-bottom: 1px solid rgb(var(--venia-border));
+    display: flex;
+    justify-content: space-between;
```
Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/577f1a335eac9fd566f387945ff957ae07298680.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.css:
```diff
@@ -0,0 +1,64 @@
+
+.header {
+    border-bottom: 1px solid rgb(var(--venia-border));
+    display: flex;
```
Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/577f1a335eac9fd566f387945ff957ae07298680.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.js:
```diff
@@ -0,0 +1,55 @@
+        return (
+            <span className={classes.error}>
+                {`Error loading shipping methods.
+                Please ensure a shipping address is set and try again.`}
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/66e73bb7e331baccf333e322f809fa28c4e5636d.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.js:
```diff
@@ -0,0 +1,55 @@
+                radio: classes.radio,
+                radioLabel: classes.radioContents,
+                root: classes.root
+            }}
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/64337d6f2440f5cbf82aaf8c5f2c8179c46139d8.

@tjwiebell fyi, I had to touch Cart page `shippingMethods` here too. Mostly just incorporating the overrides you had there into the base `RadioGroup` component itself.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.css:
```diff
@@ -0,0 +1,46 @@
+.root {
+    align-items: flex-start;
+    display: flex;
+    flex-direction: column;
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/5e349366657a3eb428eb2afcc9161fa60dd3d5a0.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.css:
```diff
@@ -0,0 +1,46 @@
+.loadingRoot {
+    composes: root from '../../LoadingIndicator/indicator.css';
+    height: unset;
+    margin: unset;
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/366a9f8c75ec2a2659f5a40c6df21ff16f73af3f.

This gave me enough room to add a loading message too 💪 

Desktop:
![Screen Shot 2020-04-07 at 9 58 13 AM](https://user-images.githubusercontent.com/13182778/78685614-9c02d100-78b7-11ea-8f4f-61853bbd0926.png)

Mobile:
![Screen Shot 2020-04-07 at 9 58 27 AM](https://user-images.githubusercontent.com/13182778/78685630-a0c78500-78b7-11ea-93c6-50c6fc99993a.png)

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.css:
```diff
@@ -0,0 +1,46 @@
+.radio {
+    display: grid;
+    gap: 0.5rem;
+    grid-template-columns: max-content auto;
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/64337d6f2440f5cbf82aaf8c5f2c8179c46139d8.

supernova-at commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.js:
```diff
@@ -1,46 +1,112 @@
+    }
+
+    return (
+        <>
```
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/3f7f9d788e5aaa8616a2c5d33c2a82a2b79e31d8.

jimbo commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingRadios.css:
```diff
@@ -10,7 +10,7 @@
 .radioLabel {
     display: grid;
-    grid-template-columns: min-content auto;
+    grid-template-columns: minmax(auto, max-content) auto;
```
I believe `minmax(auto, max-content)` is identical to `minmax(auto, auto)`, which is identical to `auto`. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/minmax).

> **`auto`**
>
> As a maximum, identical to max-content.

supernova-at replied:
Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/40211e9bf02e4242fea36e8136aa78b4d2ea8d90.

supernova-at commented:
Note to self:
- [ ] Include `classes` in `propType` definitions
- [ ] Double check `RadioGroup` on cart page and when multiple options are rendered

devops-pwa-codebuild commented:
**Performance Test Results**

The following fails have been reported by WebpageTest

https://pr-2280.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.69

supernova-at commented:
PR Updated for QA:
  - [x] Sort shipping methods by price (https://github.com/magento/pwa-studio/pull/2280/commits/47de8b52aaf0c7637e95ce9cdc7d2076bc038ea5)
  - [x] UpdateModal clears form state on cancel (pill button or "x" icon) (https://github.com/magento/pwa-studio/pull/2280/commits/eddcf8918b4bc1ad1f3b12b9acd8410f7603c0cc)
  - [x] Fixes Safari bug where the UpdateModal was not scrolling properly (https://github.com/magento/pwa-studio/pull/2280/commits/e3bdbbbcdf724a88bb6266ddbd2870e0e78163bd)
  - [x] Properly updates the PriceSummary when a new shipping method is selected (updated) (https://github.com/magento/pwa-studio/pull/2280/commits/ecc7d170544a228a2ec96056c82c0211e0ade438)

supernova-at commented:
PR Updated:
  * Changed `useResetForm` to be a hook instead of a talon ( )
  * Fixed a bug where the Price Summary total wasn't getting properly updated when selecting or updating the shipping method. Updated some query logic to re-use Price Summary fragments instead of hardcoding them ( )

@dpatil-magento for some reason the "Fixed" shipping method displays $10 in the UI but only $5 in the price summary. All of the other shipping methods work correctly. I have no idea why this one is different 😬 . The client is just reflecting back the data from the server. This may be a GraphQL bug?

Screenshots:

![Screen Shot 2020-04-10 at 10 18 16 AM](https://user-images.githubusercontent.com/13182778/79001946-0d8b8b00-7b15-11ea-80fe-cf3693bfca39.png)

![Screen Shot 2020-04-10 at 10 17 40 AM](https://user-images.githubusercontent.com/13182778/79001957-12e8d580-7b15-11ea-8fec-73df639ba9b1.png)

dpatil-magento commented:
@supernova-at That's known graphql bug, https://jira.corp.magento.com/browse/MC-31420 :)

sirugh commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.gql.js:
```diff
@@ -36,11 +38,20 @@ export const SET_SHIPPING_METHOD = gql`
+                ...ShippingSummaryFragment
+                prices {
+                    ...GrandTotalFragment
+                }
```
On the cart page we approach this a bit differently. In [that mutation](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingRadios.js#L71) we re-fetch the entire cart data. I prefer this more granular approach, and Apollo's own docs even suggest that you should be specific about the data you are updating.

With that said, I think it is a good idea to use the entire `priceSummaryFragment` here rather than specific pieces of it.

supernova-at replied:
Good call. Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/3d0cadee0db37574495b52671a12be158ab36fea.

tjwiebell commented on packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/EditForm/editForm.gql.js:
```diff
@@ -1,6 +1,7 @@
 import gql from 'graphql-tag';
 
 import { ShippingInformationFragment } from '../shippingInformationFragments.gql';
+import { ShippingMethodsFragment } from '@magento/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethodFragments.gql';
```
Whoops, fix import to relative.

supernova-at replied:
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/4622f28ad800516b494c3fa5ff7a8bba6f01a9ee.

dpatil-magento commented:
QA Pass, need review on latest commits.

jimbo commented on packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModalCancelIconButton.js:
```diff
@@ -12,7 +12,7 @@ const UpdateModalCancelIconButton = props => {
 
     return (
-        <button className={className} onClick={handleClick}>
+        <button className={className} onClick={handleClick} type="button">
```
So important! The default `type` is `submit`, and we listen for the form's `submit` event rather than just the submit button's `click` event. 😁
